### PR TITLE
feat: add erc20 paymaster test and refactor testing

### PR DESCRIPTION
**Pull request by sarahschwartz**
This PR:
- adds a test for the custom erc20 paymaster tutorial
- adds types for each supported test action
- refactors the testing framework to use a config file for the step details instead of putting everything in the `span` element. This way the details of each step is type checked.

### CODE REVIEW

File: content/tutorials/erc20-paymaster/10.index.md [hunk 1]
```diff
@@ -40,19 +40,26 @@ Atlas is a smart contract IDE that lets you write, deploy, and interact with con
...
-1. Navigate into the project directory:
+This creates a new ZKsync Era project called `custom-paymaster-tutorial` with a basic `Greeter` contract.
 
-    ```sh
-    cd custom-paymaster-tutorial
-    ```
+Next, navigate into the project directory and install the dependencies:
+
+<span id="wait-for-init" data-test-action></span>
+<span id="npm-install" data-test-action></span>
```
Comment (itsacoyote) [inline content/tutorials/erc20-paymaster/10.index.md:56]: Could we possibly use a vue component to make adding these test tags easier to write with props, etc? Just a thought, I'm not sure if that affects the test process being a vue component.

**Review by itsacoyote — APPROVED** [review]
Just a curious question, otherwise looks good to me!

File: content/tutorials/erc20-paymaster/10.index.md (hunk 1 above)
Comment (sarahschwartz) [reply content/tutorials/erc20-paymaster/10.index.md:56]: Yes!

**Review by itsacoyote — APPROVED** [review]
🎉

**Pull request merged**